tools/clang/test/HLSL/write-const-arrays.hlsl

```diff
@@ -13,38 +13,38 @@ StructuredBuffer<uint> g_robuf;
 Texture1D<uint> g_tex;
 uint g_cbuf[4];
 
-const groupshared uint gs_val[4];
+const groupshared uint gs_val[4];                           /* fxc-error {{X3012: 'gs_val': missing initial value}} */
```

Comment:
I also question whether there is value in having this file tested by FXC since it is just a mess of warnings, and since we're not actively developing FXC any regression here is of limited utility.

Reply:
Changes moved to #5148.

Even this particular error highlights where DXC is lacking.  A `const groupshared` global *should* have an initializer, otherwise it has no way to be used without reading undef values.  I believe DXC's support for initializers on `groupshared` globals is also missing (unless that was fixed more recently).

Reply:
We should discuss this in the other PR #5148, but we have an FXC exclusion list for tests we don't want to try on FXC.  I think this file is still valuable to compare with FXC.  Some of this "mess" stems from this difference in DXC allowing uninitialized const variables to be declared without an initializer.  For FXC, because it's less resilient to prior errors, any line that causes an error to be emitted is removed before compiling again and looking for more errors later in the test.  That means if you use the decl that produced an error on FXC, that use will also produce an error.

Reply:
I wonder if there is a better way for us to contrast then diagnostic differences between FXC and DXC than having our test cases marked up with FXC error and warning text. I don't really have an idea here, but I'm curious if we can do something.